app/models/text_field_type.rb

```diff
@@ -1,8 +1,11 @@
+require 'jsonb_accessor'
 class TextFieldType < FieldType
   attr_accessor :text
+  jsonb_accessor :data, text: :string
 
   validates :text, presence: true, if: :validate_presence?
   validate :text_length, if: :validate_length?
+  validate :text_unique, if: :validate_uniqueness?
```

Comment:
We're currently getting it working with presence - if that can work with our unique setup then uniqueness should as well. Look in cortex how we're implementing that and recreate the process of uniqueness